lib/trino-filesystem/src/main/java/io/trino/filesystem/Locations.java

```diff
@@ -29,6 +33,53 @@ public static String appendPath(String location, String path)
         return location + path;
     }
 
+    public static URI toUri(String location)
+    {
+        validateLocation(location);
+
+        String path = location;
+        String scheme = null;
+        int colonIndex = location.indexOf(":");
+        if (colonIndex != -1) {
+            scheme = location.substring(0, colonIndex);
+            path = location.substring(colonIndex + 1);
+        }
+        try {
+            return new URI(scheme, null, path, null, null);
+        }
+        catch (URISyntaxException e) {
+            throw new IllegalArgumentException("Invalid location: " + location, e);
+        }
+    }
+
+    public static String getParent(String location)
+    {
+        validateLocation(location);
+
+        int lastIndexOfSlash = location.lastIndexOf('/');
+        if (lastIndexOfSlash > 0) {
+            String parent = location.substring(0, lastIndexOfSlash);
+            if (!parent.endsWith("/") && !parent.endsWith(":")) {
+                return parent;
+            }
+        }
+        throw new IllegalArgumentException(format("Location [%s] does not have parent", location));
+    }
+
+    public static String getFileName(String location)
+    {
+        validateLocation(location);
+
+        return location.substring(location.lastIndexOf('/') + 1);
```

Comment:
This returns an empty string for a location like `s3://foo/bar/`. Is that what we want?

Reply:
`public static Optional<String> getFileName(String location)`
another alternative

Reply:
That seems fine. I'm just confirming you considered this.